scripts/shared/lib/deploy_bundle

```diff
@@ -0,0 +1,249 @@
+  if [[ -z "${VERSION}" ]] ||
+   [[ -z "${SUBM_NS}" ]] ||
+   [[ -z "${CHANNEL}" ]]; then
+    echo "[ERROR] Required environment variables not loaded"
+    echo "    VERSION"
+    echo "    SUBM_NS"
+    echo "    CHANNEL"
+    exit 3
+  fi
+
+  # Create new namespace
+  run_subm_clusters create_submariner_namespace
+
+  # Create the custom catalog source
+  run_subm_clusters create_catalog_source
+
+  # Install the submariner operator bundle
+  run_subm_clusters install_bundle
+}
+
+function create_submariner_namespace {
+  echo "[INFO](${cluster}) Create the namespace..."
+  kubectl create namespace "${SUBM_NS}" || :
+}
+
+function create_catalog_source() {
+  echo "[INFO](${cluster}) Create the catalog source..."
+
+  kubectl delete catalogsource/submariner-catalog-source -n "${MARKETPLACE_NAMESPACE}" --wait --ignore-not-found
+
+  # Create the CatalogSource
+cat <<EOF | kubectl apply -f -
+apiVersion: operators.coreos.com/v1alpha1
+kind: CatalogSource
+metadata:
+  name: submariner-catalog-source
+  namespace: ${MARKETPLACE_NAMESPACE}
+spec:
+  sourceType: grpc
+  image: ${INDEX_IMG}
+  displayName: Submariner Catalog Source
+  publisher: Submariner.io (Test)
+  updateStrategy:
+    registryPoll:
+      interval: 5m
+EOF
+
+  # Wait for the catalogSource Readiness
+  if ! (timeout 5m bash -c "until [[ $(kubectl --context="${cluster}" get catalogsource -n ${MARKETPLACE_NAMESPACE} submariner-catalog-source -o jsonpath='{.status.connectionState.lastObservedState}') -eq 'READY' ]]; do sleep 10; done"); then
+      echo "[ERROR] CatalogSource is not ready."
+      exit 1
+  fi
+
+  # Debug
+  kubectl -n ${MARKETPLACE_NAMESPACE} get catalogsource --ignore-not-found
+  kubectl -n ${MARKETPLACE_NAMESPACE} get pods --ignore-not-found
+  kubectl -n ${MARKETPLACE_NAMESPACE} get packagemanifests --ignore-not-found | grep 'Submariner Catalog Source' || true
+
+
+  # Create the OperatorGroup
+cat <<EOF | kubectl apply -f -
+apiVersion: operators.coreos.com/v1
+kind: OperatorGroup
+metadata:
+  name: submariner-group
+  namespace: ${SUBM_NS}
+spec:
+  targetNamespaces:
+    - ${SUBM_NS}
+EOF
+
+  # Debug
+  kubectl get og -n "${SUBM_NS}" --ignore-not-found
+
+  echo "[INFO](${cluster}) Catalog source created"
+}
+
+function install_bundle() {
+  local installPlan
+
+  # Delete previous catalogSource and Subscription
+  kubectl delete sub/submariner-subscription -n "${SUBM_NS}" --wait --ignore-not-found
+
+  # Create the Subscription (Approval should be Manual not Automatic in order to pin the bundle version)
+  echo "[INFO](${cluster}) Install the bundle..."
+cat <<EOF | kubectl apply -f -
+apiVersion: operators.coreos.com/v1alpha1
+kind: Subscription
+metadata:
+  name: submariner-subscription
+  namespace: ${SUBM_NS}
+spec:
+  channel: ${CHANNEL}
+  installPlanApproval: Manual
+  name: ${OPERATOR_NAME}
+  source: submariner-catalog-source
+  sourceNamespace: ${MARKETPLACE_NAMESPACE}
+  startingCSV: ${OPERATOR_NAME}.v${VERSION}
+EOF
```

Comment:
This would be better if it was in a real YAML file which gets validated. You can use `render_template` function from `clusters.sh`